healthcheck/src/healthcheck/cache.py

```diff
@@ -0,0 +1,58 @@
+from collections.abc import Awaitable, Callable
+from functools import wraps
+from typing import ParamSpec, TypeVar
+
+from diskcache import FanoutCache
+
+from healthcheck.constants import CACHE_KEY_PREFIX, CACHE_LOCATION, DEFAULT_CACHE_TTL
+
+P = ParamSpec("P")
+R = TypeVar("R")
+
+_cache: FanoutCache | None = None
+
+
+def init_cache() -> FanoutCache:
+    """Get or create the disk cache instance."""
+    global _cache  # noqa: PLW0603
+    if _cache is None:
+        _cache = FanoutCache(CACHE_LOCATION)
+    return _cache
+
+
+def close_cache() -> None:
+    """Close the disk cache instance."""
+    global _cache  # noqa: PLW0603
+    if _cache is not None:
+        _cache.close()
+        _cache = None
+
+
+def memoize(
+    key: str, ttl: float = DEFAULT_CACHE_TTL
```

Comment:
Not a fan of the `ttl` abbreviation because we'll use it in a request context where TTL has a different meaning in network than _expiration_